komfydb/storage/table_iterator.cc

```diff
@@ -0,0 +1,56 @@
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+#include "komfydb/common/tuple.h"
+
+namespace komfydb::storage {
+
+TableIterator::TableIterator(TransactionId tid, int table_id)
+    : tid(tid), table_id(table_id) {}
+
+absl::StatusOr<bool> TableIterator::LoadNextPage() {
+  if (++page_ctr >= num_pages)
+    return false;
+    ASSIGN_OR_RETURN(Page* page, bufferpool->GetPage(PageId(table_id, page_ctr));
+    tuples = page->GetTuples();
+    current_tuple = tuples.begin();
+    return true;
+}
+
+absl::Status TableIterator::LoadFirstPage() {
+  page_ctr = -1;
+  ASSIGN_OR_RETURN(bool good, LoadNextPage());
+  if (!good)
+    return absl::OutOfRangeError("No pages in this file/table");
+  return absl::StatusOK();
+}
+
+absl::Status TableIterator::Open() {
```

Comment:
Maybe we can use RAII here. Instead of having initialization function like this we may have a static factory function `static absl::StatusOr<std::unique_ptr<TableIterator>> TableIterator::Create`. Then we don't need the `TableIterator::Close()` as we can move the closing logic to the destructor.